### ISSUE TITLE
Documentation and post processing

### ISSUE THREAD

**Issue by Sina-av** [hugary1995/raccoon#102]
Hey @hugary1995!
I was going over the code and found some points, thought let you know:
1) In finding the Eigen strains , it seems that the comment is missing an inverse, meaning it should be \\\mathbb{C}^-1 https://github.com/hugary1995/raccoon/blob/5ed2d7e3c89e3643a51b59e47a2806713078f2ac/src/materials/ComputeEigenstrainFromFunctionInitialStress.C#L14-L17
2) (This one is not really important) In this one, the comment should be "the plasticity model"
https://github.com/hugary1995/raccoon/blob/5ed2d7e3c89e3643a51b59e47a2806713078f2ac/include/materials/small_deformation_models/ComputeSmallDeformationStress.h#L33-L34
3) Also for the post processing purposes, in branch "devel", there is a class for calculating fracture energy. Is there any reason that class is not in the main branch? I'm thinking about a class that add the strain energy to the fracture energy in the Postprocess section (which can also be used later as a convergence criterion for solving the system of equations.)
I also went over PR #27, on how to use ElasticEnergyDensity material already available in moose. But I think that one does not work with AD variables, and other that that we already have the strain energy (or energies) from our elastic (and plastic) materials so no need to calculate them again. So maybe we can have a post process class of type "ElementIntegralPostprocessor" which gets strain energy and crack surface density and fracture toughness and returns the energy. 
Please let me know if this is a good idea.
I can also make a PR for what I'm proposing if that is easier to follow.
Thanks

**Comment by hugary1995** [comment]
1. Good catch, it should be the inverse. Actually, to be more precise, I remember this class is assuming isotropic elasticity, so I took some short cuts in the code. I will clarify that.
2. Good catch again. I'll correct. I'll also do a global search to see if there are other typos like this.

Regarding your third point, I get what you mean, and I mostly agree with you. The canonical way of computing element integrals is indeed using an element integral post-processor
```
[Postprocessors]
  [some_energy]
    type = ADElementIntegralMaterialProperty
    mat_prop = name_of_the_mat_prop
  []
[]
```
which integrates a material property defined per quadrature point called `name_of_the_mat_prop` over the specified block.

That material property can come from a variety of objects. In raccoon, some common names are `psie` for the strain energy density, `psie_active` for the active strain energy density, `psip` for the plastic energy density, and `psip_active` for the active plastic energy density. If you just substitute the name of the quantity you want to integrate into that `ADElementIntegralMaterialProperty`, it ought to work out of the box.

However, there are some quantities that are not computed per quadrature point by default, like the fracture energy you mentioned. To do that, I have provided an object that calculates the crack surface density:
```
[Materials]
  [crack_surface_density]
    type = CrackSurfaceDensity
    phase_field = d
  []
[]
```
which by default calculates the crack surface density and stores it as a material property called `gamma`. Then you can use the `ADElementIntegralMaterialProperty` to integrate that to get the total crack surface area.

You can extend that capability to compute the fracture energy density using a parsed material:
```
[Materials]
  [crack_surface_density]
    type = CrackSurfaceDensity
    phase_field = d
  []
  [fracture_energy_density]
    type = ADParsedMaterial
    f_name = psif
    function = 'Gc * gamma'
    material_property_names = 'Gc gamma'
  []
[]

[Postprocessors]
  [fracture_energy]
    type = ADElementIntegralMaterialProperty
    mat_prop = psif
  []
[]
```

Hope this helps, and let me know if you have further questions. I will address your points 1 and 2.

**Comment by Sina-av** [comment]
Thanks @hugary1995  for the comments. Yes, that is really helpful and is what I meant. I now also think that 'ADParsedMaterial' is a cleaner way to find the fracture energy as we already have gamma (instead of making a new class). Also to address the total energy (without using a new class):
Let's say we have a ductile fracture where in Materials block we define 'elasticity' to get 'psie', 'plasticity' to get 'psip', and 'fracture_energy_density' to get 'psif'. 
Then can we use
``` 
[Materials]
...
[total_energy]
   type = ADDerivativeSumMaterial
    f_name = psie_psip_psif
    sum_materials = 'psie psip psif'
    args = 'disp_x'
    outputs = all 
  [] 
...
[]
```
to find the total energy in each quadrature point? 
And later on, use 'ADElementIntegralMaterialProperty' to post process it:
```
[Postprocessors]
  [total_energy]
    type = ADElementIntegralMaterialProperty
    mat_prop = psie_psip_psif
  []
[]
```
Thanks.

**Comment by hugary1995** [comment]
Yes, that should work!

**Comment by hugary1995** [comment]
Closed by #103
